test/k8sT/Updates.go

```diff
@@ -87,13 +87,14 @@ var _ = Describe("K8sValidatedUpdates", func() {
 	It("Updating Cilium stable to master", func() {
 		By("Creating some endpoints and L7 policy")
 		kubectl.Apply(demoPath).ExpectSuccess()
-		_, err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", timeout)
-		Expect(err).Should(BeNil())
 
-		kubectl.Exec(fmt.Sprintf("%s scale deployment %s --replicas=10",
+		kubectl.Exec(fmt.Sprintf("%s scale deployment %s --replicas=5",
 			helpers.KubectlCmd, helpers.App1)).ExpectSuccess(
 			"Cannot scale %v deployment", helpers.App1)
 
+		_, err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", timeout)
```

Comment:
Doesn't this expect the Cilium endpoints to already have been created and then waits for the *already created* endpoints to become ready?

I think we need a helper that ensures that:
1. the number of pods with the corresponding labels have been created
2. corresponding cilium endpoints exist in the required number
3. the endpoins are ready

Reply:
After `WaitForPods` we wait in all cilium pods that the endpoints are in ready state. 

https://github.com/eloycoto/cilium/blob/fbacd896218b1b855606cae666e0f390146f2a2c/test/k8sT/Updates.go#L102

Do you want something like `CiliumPolicyAction` for WaitforEndpoints?

Reply:
Yes, but doesn't it assume that all the endpoints are already created and known to Cilium? If you scale, the pods are not even created yet.

Reply:
OK